src/main/webapp/app/shared/markdown-editor/select-with-search/select-with-search.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output, ViewChild } from '@angular/core';
+import { InteractiveSearchCommand } from 'app/shared/markdown-editor/commands/interactiveSearchCommand';
+import { AlertService } from 'app/core/util/alert.service';
+import { onError } from 'app/shared/util/global.utils';
+import { Subject, debounce, distinctUntilChanged, switchMap, takeUntil, timer } from 'rxjs';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { MatMenuItem, MatMenuTrigger } from '@angular/material/menu';
+
+interface SearchQuery {
+    searchTerm: string;
+    noDebounce: boolean;
+}
+
+@Component({
+    selector: 'jhi-select-with-search',
+    templateUrl: './select-with-search.component.html',
+    styleUrls: ['./select-with-search.component.scss'],
+})
+export class SelectWithSearchComponent implements OnInit, OnDestroy {
+    @Input() command: InteractiveSearchCommand;
+    @Output() valueSelect = new EventEmitter<any | undefined>();
+
+    @ViewChild(MatMenuTrigger) menuTrigger: MatMenuTrigger;
+    @ViewChild('searchInputForm') searchInputForm: MatMenuItem;
+    @ViewChild('searchInput') searchInput: ElementRef;
+
+    private ngUnsubscribe = new Subject<void>();
+    private readonly search$ = new Subject<SearchQuery>();
+
+    values: any[] = [];
+    selectedValue: any;
```

Comment:
Is there no way to replace the `any` with string or something like that?

Reply:
I added a new interface for it